File: sdk-go/anthropic/anthropic_test.go
package anthropic_test

import (
	"os"
	"testing"

	"github.com/hoangvvo/llm-sdk/sdk-go/anthropic"
	"github.com/hoangvvo/llm-sdk/sdk-go/internal/testcommon"
	"github.com/joho/godotenv"
)

var model *anthropic.AnthropicModel

func TestMain(m *testing.M) {
	godotenv.Load("../../.env")
	apiKey := os.Getenv("ANTHROPIC_API_KEY")
	if apiKey == "" {
		panic("ANTHROPIC_API_KEY must be set")
	}

	model = anthropic.NewAnthropicModel("claude-sonnet-5", anthropic.AnthropicModelOptions{
		APIKey: apiKey,
	})

	m.Run()
}

var reasoningOptions = []testcommon.TestCaseOption{
	testcommon.WithProfile("anthropic_adaptive_reasoning"),
}

func TestTextGeneration(t *testing.T) {
	testcommon.RunTestGroup(t, model, "text_generation")
}

func TestConversation(t *testing.T) {
	testcommon.RunTestGroup(t, model, "conversation")
}

func TestToolUse(t *testing.T) {
	testcommon.RunTestGroup(t, model, "tool_use")
}

func TestStructuredOutput(t *testing.T) {
	testcommon.RunTestGroup(t, model, "structured_output")
}

func TestGenerationOptions(t *testing.T) {
	testcommon.RunTestGroup(t, model, "generation_options")
}

func TestSourceInput(t *testing.T) {
	testcommon.RunTestGroup(t, model, "source_input")
}

func TestMultimodalToolResult(t *testing.T) {
	testcommon.RunTestGroup(t, model, "multimodal_tool_result")
}

func TestWebSearch(t *testing.T) {
	testcommon.RunTestGroup(t, model, "web_search", testcommon.WithProfile("anthropic_web_search"))
}

func TestImageInput(t *testing.T) {
	testcommon.RunTestGroup(t, model, "image_input")
}

func TestReasoning(t *testing.T) {
	testcommon.RunTestGroup(t, model, "reasoning", reasoningOptions...)
}

func TestReasoningToolUse(t *testing.T) {
	testcommon.RunTestGroup(t, model, "reasoning_tool_use")
}

func TestAnthropicRefusal(t *testing.T) {
	testcommon.RunTestGroup(t, model, "anthropic_refusal")
}

File: sdk-rust/tests/anthropic_test.rs
mod common;
use llm_sdk::anthropic::*;
use std::{env, error::Error, sync::OnceLock};
use tokio::test;

fn anthropic_api_key() -> &'static String {
    static KEY: OnceLock<String> = OnceLock::new();

    KEY.get_or_init(|| {
        dotenvy::dotenv().ok();
        env::var("ANTHROPIC_API_KEY").expect("ANTHROPIC_API_KEY must be set")
    })
}

fn anthropic_model() -> AnthropicModel {
    AnthropicModel::new(
        "claude-sonnet-5".to_string(),
        AnthropicModelOptions {
            api_key: anthropic_api_key().clone(),
            ..Default::default()
        },
    )
}

test_group!(anthropic_model(), text_generation);
test_group!(anthropic_model(), conversation);
test_group!(anthropic_model(), tool_use);
test_group!(anthropic_model(), structured_output);
test_group!(anthropic_model(), generation_options);
test_group!(anthropic_model(), source_input);
test_group!(anthropic_model(), multimodal_tool_result);
test_group!(
    anthropic_model(),
    web_search,
    Some(crate::common::cases::RunTestCaseOptions {
        profile: Some("anthropic_web_search"),
    })
);
test_group!(anthropic_model(), image_input);
test_group!(
    anthropic_model(),
    reasoning,
    Some(crate::common::cases::RunTestCaseOptions {
        profile: Some("anthropic_adaptive_reasoning"),
    })
);
test_group!(anthropic_model(), reasoning_tool_use);
test_group!(anthropic_model(), anthropic_refusal);
